### PR TITLE
CG - Pull Request

### PR DESCRIPTION
**Pull request by Charley-Gooch**
my pull request my dudes

### CODE REVIEW

**Comment by EdwardAndress** [comment]
Hey Charley,

Nice work!

I'm going to leave a few comments inline against your code.  Let me know if you have any questions about them.

File: lib/Airport.rb [hunk 1]
```diff
@@ -0,0 +1,45 @@
+class Airport
+  @airportName = nil
```
Comment (EdwardAndress) [inline lib/Airport.rb]: You don't need to assign nil to these instance vars here.

Comment (EdwardAndress) [reply lib/Airport.rb]: The names become slightly tautologous (you end up doing `airport.airportName` but `airport.name` would be equally clear).

File: lib/Airport.rb [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  @airportId = nil
+  @airportPlaneCapacity = nil
+  @airportPassangerCapacity = nil
+  @airportWeather = nil
+    
+  def initialize(airportName, airportId, airportPlaneCapacity, airportPassangerCapacity, airportWeather = "clear skies")
```
Comment (EdwardAndress) [inline lib/Airport.rb]: Once you have more than a couple of parameters, it's worth considering the use of named parameters.

```
def initialize(name: name, id: id, plane_capacity: plane_capacity, passenger_capacity: passenger_capacity, weather: 'clear skies')
```

The advantage is that when you call `.new` you don't need to remember in what order the parameters should provided.

File: lib/Main.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
+require_relative "Plane"
```
Comment (EdwardAndress) [inline lib/Main.rb]: This is useful but we'd normally use a feature test here.

File: lib/Airport.rb (hunk 2 above)
Comment (Charley-Gooch) [reply lib/Airport.rb]: makes sense

File: lib/Airport.rb [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    @airportName = airportName
+    @airportId = airportId
+    @airportPlaneCapacity = airportPlaneCapacity
+    @airportPassangerCapacity = airportPassangerCapacity
+    @airportWeather = airportWeather
+  end
+
```
[the remark this answers is not in the document]
Comment (Charley-Gooch) [reply lib/Airport.rb]: i'm simply used to using these from Java
a better language

File: lib/Airport.rb [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  def setAirportPlaneCapacity(capcity)
+    @airportPlaneCapacity = capcity
+  end
+  
+  def setAirportWeather(weather)
+    @airportWeather = weather
+  end
+  
+#Getters
+  def getAirportId()
```
[the remark this answers is not in the document]
Comment (Charley-Gooch) [reply lib/Airport.rb]: again java syntax

**Comment by EdwardAndress** [comment]
There are some nice things in here and you've clearly picked up Ruby very quickly.  Each time you learn a new language, however, it's important to also learn the idioms of that language.  Using rubocop will help a little, as will reading other people's code.  In the list of pull requests you'll see solutions from a ton of other students, in different cohorts.

As you, yourself, inferred during the code review, there are some classes here which are perhaps not entirely necessary at this point in the application's development.

Finally... where are the tests?

**Pull request closed without merging**